resources/assets/components/Table/Row.js

```diff
@@ -1,16 +1,80 @@
 import React from 'react';
+import { map } from 'lodash';
 
 class Row extends React.Component {
+  constructor() {
+    super();
+
+    this.createCampaignRow = this.createCampaignRow.bind(this);
+    this.createEventRow = this.createEventRow.bind(this);
+    this.createCell = this.createCell.bind(this);
+  }
+
+  createCampaignRow(campaign) {
+    const row = [
+      {
+        url: campaign ? `/campaigns/${campaign.id}` : '/campaigns',
+        title: campaign ? campaign.title : 'Campaign Not Found',
+      },
+      {
+        url: null,
+        title: campaign ? campaign.pending_count : 0,
+      },
+      {
+        url: campaign ? `/campaigns/${campaign.id}/inbox` : '/campaigns',
+        title: 'review',
+      }
+    ];
+
+    return row;
+  }
+
+  createEventRow(event) {
+    const newQuantity = event.content.quantity ? event.content.quantity : event.content.quantity_pending;
+
+    const row = [
+      {
+        url: null,
+        title: newQuantity,
+      },
+      {
+        url: null,
+        title: event.content.why_participated,
+      },
+      {
+        url: null,
+        title: event.content.updated_at,
+      },
+      {
+        url: null,
+        title: event.user,
+      }
+    ];
+
+    return row;
+  }
+
+  createCell(data) {
+    if (data.url) {
+      return <td className="table__cell"><a href={cell.url}>{cell.title}</a></td>
+    }
+
+    return <td className="table__cell">{cell.title}</td>
+  }
+
   render() {
-    const { campaign } = this.props;
-    const campaignUrl = this.props.campaign_id ? `/campaigns/${this.props.campaign_id}` : '/campaigns';
-    const inboxUrl = this.props.campaign_id ? `/campaigns/${this.props.campaign_id}/inbox` : '/campaigns';
+    const content = this.props.type === 'campaigns' ? this.createCampaignRow(this.props.data) : this.createEventRow(this.props.data);
```

Comment:
This line makes me think we should split this up into separate `CampaignRow` and `EventRow` components, and `Row` should just be responsible for generally formatting tabular data.

Reply:
this is a great call! I just updated :)